[full-ci] Accept OCM to groups

michielbdejong opened the pull request:
## Description
<!--- Describe your changes in detail -->
These are the hooks we need in the core repo to allow the [FederatedGroups app](https://github.com/SURFnet/rd-sram-integration/tree/main/federatedgroups) to deal with outgoing and incoming OCM shares to group.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes #40530

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

phil-davis committed:
Run some CI pipelines with Ubuntu 22

phil-davis committed:
Enhance share copy-overwrite test scenarios to demonstrate current behavior

phil-davis committed:
Demonstrate copy file over shared file and folder over shared folder

phil-davis committed:
Update Symfony from major version 4 to 5

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/core?pullRequest=40589) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/core?pullRequest=40589) before we can accept your contribution.<br/>**1** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: michielbdejong<br/>:x: michiel-testing<br/>:x: navid-dada<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/core?pullRequest=40589) it.</sub>

michielbdejong commented on apps/files_sharing/js/external.js:
```diff
@@ -142,16 +142,20 @@
 									// Delete
 									$.ajax({
-										url: OC.generateUrl('/apps/files_sharing/api/externalShares/'+share.id),
+										url: OC.generateUrl('/apps/files_sharing/api/externalShares/'+share.id+"?share_type="+shareType),
```
No, we will be storing OCM-to-group shares in a different database table than OCM-to-user shares, because @jvillafanez wanted to keep the impact on the existing database table minimal.

So now, on servers where OCM-to-group is enabled, you need to include the OCM share type to get a unique share id.

DeepDiver1975 replied:
:see_no_evil:

michielbdejong commented on apps/federatedfilesharing/lib/Controller/OcmController.php:
```diff
@@ -284,6 +301,17 @@ public function processNotification(
 	) {
+		// Allow the Federated Groups app to overwrite the behaviour of this endpoint
+		if (\OC::$server->getAppManager()->isEnabledForUser('federatedgroups')) {
+			$controller = \OCA\FederatedGroups\Application::getOcmController($this->request);
```
And then if the event is picked up the default code is not ran? What if the event is picked up twice?

michielbdejong replied:
I guess it could work with something like the PHP equivalent of the [stopImmediatePropagation pattern](https://developer.mozilla.org/en-US/docs/Web/API/Event/stopImmediatePropagation) which I know from the browser world, but it still feels like using a config variable that is the name of a class, like in https://github.com/owncloud/core/blob/838d8819411aee8d661e2b29514efca6d101eefc/apps/files_sharing/lib/Controller/ShareesController.php#L385-L389 is a better approach here.

michielbdejong commented on apps/federatedfilesharing/lib/Controller/OcmController.php:
```diff
@@ -177,6 +177,23 @@ public function createShare(
 	) {
+		// Allow the Federated Groups app to overwrite the behaviour of this endpoint (but only for group shares)
+		if (\OC::$server->getAppManager()->isEnabledForUser('federatedgroups') && ($shareType === 'group')) {
+			$controller = \OCA\FederatedGroups\Application::getOcmController($this->request);
```
Noted! We'll change it.

michielbdejong replied:
Btw, the reference to "the other PR" is [this discussion](https://github.com/owncloud/core/pull/40577#discussion_r1085438465)

rizlas committed:
Handle X-Robots-Tag header valid values other than none

rizlas committed:
Add changelog. Precise messages in setupchecks securityHeaders

rizlas committed:
Changelog

rizlas committed:
Fix messages consistency with checkGeneric tests

rizlas committed:
Fix check header x-robots-tag exists

rizlas committed:
Update changelog/unreleased/40715

Co-authored-by: ho4ho <42564859+ho4ho@users.noreply.github.com>

phil-davis committed:
Update changelog/unreleased/40715

phil-davis commented:
@michielbdejong https://drone.owncloud.com/owncloud/core/38230/3/7
It still fails code-style checks. If you fix those very minor things then we will get to see the results of unit tests and acceptance tests.

https://drone.owncloud.com/owncloud/core/38230/6/8
You might also need to add some "ignores" to `phpstan.neon` for the `Access to an undefined property` messages.

michielbdejong commented on apps/files_sharing/lib/Controller/Share20OcsController.php:
```diff
@@ -552,7 +552,7 @@ public function createShare() {
 				$share->setPassword($password);
 			}
-		} elseif ($shareType === Share::SHARE_TYPE_REMOTE) {
+		} elseif ($shareType === Share::SHARE_TYPE_REMOTE || $shareType === Share::SHARE_TYPE_REMOTE_GROUP) {
```
@navid-dada can you check if this one is needed? It looks like maybe this is code that deals with OCS and not OCM

michielbdejong replied:
Let's try without this change and we can add it back if we do find we need it.

michielbdejong replied:
As discusses with @navid-dada, yes it's needed.

phil-davis commented on apps/federatedfilesharing/lib/Controller/OcmController.php:
```diff
@@ -306,7 +306,7 @@ public function processNotification(
 		if ($controllerClass !== '') {
 			$controller = \OC::$server->query($controllerClass);
-		  return $controller->processNotification(
+		  	return $controller->processNotification(
```
This line has tab, tab, space, space, tab - the 2 spaces are what is causing code-style to complain.

ownclouders commented:
:boom: Acceptance tests pipeline <strong>apiShareUpdateToRoot-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/38252/68
<!-- id: pr-40589 -->

rizlas committed:
if statement styling. Increase browser compatibility. Increase security check readability.

rizlas committed:
fix line indentation

jvillafanez commented:
Just saying that the commits should be squashed when the PR is ready to merge.

michielbdejong committed:
Accept OCM to groups

Signed-off-by: Michiel de Jong <michiel@unhosted.org>

michielbdejong committed:
Accept OCM to groups

All Broken tests are fixed

Signed-off-by: Michiel de Jong <michiel@unhosted.org>
Signed-off-by: navid-shokri <naviid-shokrii@gmail.com>

ownclouders committed:
[tx] updated from transifex

jvillafanez committed:
Replace unit tests for the report command

michielbdejong commented:
Hm, build was killed, I'm not sure why.

ownclouders committed:
[tx] updated from transifex

jvillafanez commented:
There are some tests failing in https://drone.owncloud.com/owncloud/core/38306/148/18

```
{"reqId":"hKNpkBk8bIqVRcbEE0R2","level":4,"time":"2023-04-19T09:20:55+00:00","remoteAddr":"192.168.28.5","user":"Alice","app":"webdav","method":"PROPFIND","url":"\/remote.php\/dav\/files\/Alice\/","message":"Exception: No share provider for share type 7: {\"Exception\":\"OC\\\\Share20\\\\Exception\\\\ProviderException\",\"Message\":\"No share provider for share type 7\",\"Code\":0,\"Trace\":\"#0 \\\/drone\\\/src\\\/lib\\\/private\\\/Share20\\\/Manager.php(160): OC\\\\Share20\\\\ProviderFactory->getProviderForType()\\n#1 \\\/drone\\\/src\\\/lib\\\/private\\\/Share20\\\/Manager.php(1272): OC\\\\Share20\\\\Manager->shareTypeToProviderMap()\\n#2 \\\/drone\\\/src\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/SharesPlugin.php(146): OC\\\\Share20\\\\Manager->getAllSharesBy()\\n#3 \\\/drone\\\/src\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/SharesPlugin.php(197): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\SharesPlugin->getSharesForNodeIds()\\n#4 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/WildcardEmitterTrait.php(89): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\SharesPlugin->handleGetProperties()\\n#5 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1052): Sabre\\\\DAV\\\\Server->emit()\\n#6 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(984): Sabre\\\\DAV\\\\Server->getPropertiesByNode()\\n#7 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1662): Sabre\\\\DAV\\\\Server->getPropertiesIteratorForPath()\\n#8 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1642): Sabre\\\\DAV\\\\Server->writeMultiStatus()\\n#9 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/http\\\/lib\\\/Sapi.php(83): Sabre\\\\DAV\\\\Server->Sabre\\\\DAV\\\\{closure}(*** sensitive parameters replaced ***)\\n#10 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(490): Sabre\\\\HTTP\\\\Sapi::sendResponse()\\n#11 \\\/drone\\\/src\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(253): Sabre\\\\DAV\\\\Server->invokeMethod()\\n#12 \\\/drone\\\/src\\\/apps\\\/dav\\\/lib\\\/Server.php(348): Sabre\\\\DAV\\\\Server->start()\\n#13 \\\/drone\\\/src\\\/apps\\\/dav\\\/appinfo\\\/v2\\\/remote.php(31): OCA\\\\DAV\\\\Server->exec()\\n#14 \\\/drone\\\/src\\\/remote.php(165): require_once('\\\/drone\\\/src\\\/apps...')\\n#15 {main}\",\"File\":\"\\\/drone\\\/src\\\/lib\\\/private\\\/Share20\\\/ProviderFactory.php\",\"Line\":138}"}
{"reqId":"hKNpkBk8bIqVRcbEE0R2","level":3,"time":"2023-04-19T09:20:55+00:00","remoteAddr":"192.168.28.5","user":"Alice","app":"PHP","method":"PROPFIND","url":"\/remote.php\/dav\/files\/Alice\/","message":"Cannot modify header information - headers already sent by (output started at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Message.php:117) at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Sapi.php#66"}
{"reqId":"hKNpkBk8bIqVRcbEE0R2","level":3,"time":"2023-04-19T09:20:55+00:00","remoteAddr":"192.168.28.5","user":"Alice","app":"PHP","method":"PROPFIND","url":"\/remote.php\/dav\/files\/Alice\/","message":"Cannot modify header information - headers already sent by (output started at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Message.php:117) at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Sapi.php#70"}
{"reqId":"hKNpkBk8bIqVRcbEE0R2","level":3,"time":"2023-04-19T09:20:55+00:00","remoteAddr":"192.168.28.5","user":"Alice","app":"PHP","method":"PROPFIND","url":"\/remote.php\/dav\/files\/Alice\/","message":"Cannot modify header information - headers already sent by (output started at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Message.php:117) at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Sapi.php#70"}
{"reqId":"hKNpkBk8bIqVRcbEE0R2","level":3,"time":"2023-04-19T09:20:55+00:00","remoteAddr":"192.168.28.5","user":"Alice","app":"PHP","method":"PROPFIND","url":"\/remote.php\/dav\/files\/Alice\/","message":"Cannot modify header information - headers already sent by (output started at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Message.php:117) at \/drone\/src\/lib\/composer\/sabre\/http\/lib\/Sapi.php#70"}
```

navid-shokri committed:
fix the broken tests

Signed-off-by: navid-shokri <naviid.shokrii@gmail.com>

ownclouders committed:
[tx] updated from transifex

navid-shokri commented:
@phil-davis 
I fixed all the tests locally but Pipelines are failing. we cannot find the root cause of this. is it an infrastructure failure or are there some errors on our side? would you be able to review the process and let us know if there are any points to regard?

phil-davis commented:
@navid-shokri I have taken a copy of this code, rebased and made PR #40749 and put `full-ci` in the title.

I will have a look at what is failing and see if I can fix it...

phil-davis committed:
Fix code style

phil-davis committed:
Fix code style

IljaN committed:
Limit fed-share token length (#40740)

phil-davis commented:
@navid-shokri @michielbdejong I have got the PHP unit tests and the acceptance tests passing.
See the extra commits in PR #40749 - cherry-pick those, or copy the code, whatever.

The JavaScript tests are also failing. You should have a look at that. I haven't looked at what JS you have touched.
https://drone.owncloud.com/owncloud/core/38320/8/4

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

dependabot[bot] committed:
Bump pear/pear-core-minimal from 1.10.11 to 1.10.13

Bumps [pear/pear-core-minimal](https://github.com/pear/pear-core-minimal) from 1.10.11 to 1.10.13.
- [Release notes](https://github.com/pear/pear-core-minimal/releases)
- [Commits](https://github.com/pear/pear-core-minimal/compare/v1.10.11...v1.10.13)

---
updated-dependencies:
- dependency-name: pear/pear-core-minimal
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

phil-davis committed:
Changelog for pear/pear-core-minimal 1.10.13

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Cleanup acceptance tests

phil-davis committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Use a shorter name in CI for mariaDB

phil-davis committed:
Add unit tests pipelines for mariaDB 10.9 10.10 10.11

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

SwikritiT committed:
make a paramater accept null values

SagarGi committed:
Added default value for skipImportLdif as false

SagarGi committed:
Added default value for occlastCode default as null

DeepDiver1975 committed:
fix: use umask (#40765)

* fix: use umask

* chore: code style changes

phil-davis committed:
Add test scenarios for Last Login on Users page

phil-davis committed:
Hide lastLogin column on Users page when disabled

phil-davis committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Cleanup code in TestHelpers

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
skip new last-login UI scenarios on old oC10

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Cleanup webUI test code

ownclouders committed:
[tx] updated from transifex

jvillafanez committed:
Add command to remove obsolete storages from the filecache

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Update vendor-bin dependencies

DeepDiver1975 committed:
fix: data length validation on UsersController (#40769)

DeepDiver1975 committed:
Automated changelog update [skip ci]

jvillafanez committed:
Provide option to show possible candidates

jvillafanez committed:
Refine DB query

esgarov committed:
.org with .com replaced

jvillafanez committed:
Adjust text

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Update to phpunit 9.6.8

ownclouders committed:
[tx] updated from transifex

Commit:
Align to new Google authorization URI (#40783)

* align to new authorization URI

* add changelog

* add links in changelog

Commit:
Automated changelog update [skip ci]

jnweiger committed:
merge Elcin's and Martin's suggestions.

ownclouders committed:
[tx] updated from transifex

jvillafanez committed:
Fix issue with postgresql

phil-davis committed:
Adjust tests for issue 40787 and 40788

phil-davis committed:
Add scenario to demonstrate copying a folder onto another folder

dependabot[bot] committed:
Bump guzzlehttp/guzzle from 7.5.1 to 7.6.1

Bumps [guzzlehttp/guzzle](https://github.com/guzzle/guzzle) from 7.5.1 to 7.6.1.
- [Release notes](https://github.com/guzzle/guzzle/releases)
- [Changelog](https://github.com/guzzle/guzzle/blob/7.6/CHANGELOG.md)
- [Commits](https://github.com/guzzle/guzzle/compare/7.5.1...7.6.1)

---
updated-dependencies:
- dependency-name: guzzlehttp/guzzle
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Changelog for guzzlehttp/guzzle 7.6.1

phil-davis committed:
Automated changelog update [skip ci]

jvillafanez committed:
Fix problems with oracle and rename showed columns

Commit:
add loginInOwnCloud to Log::

Commit:
add changelog

jvillafanez committed:
Add changelog entry

Commit:
fix style

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Always return an int for Symfony Command execute method

phil-davis committed:
Changelog for Symfony Command execute method changes

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Exit with success when signing-key has been set

phil-davis committed:
Changelog for security:sign-key:create exit status fix

phil-davis committed:
Fix 'does not exists' text

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Adjust Symfony OutputFormatterInterface in unit tests

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Adjust PHP unit tests for Symfony 5

phil-davis committed:
Bump version to 10.13.0 prealpha

phil-davis committed:
Do not try to revert LDAP settings of a deleted config

ownclouders committed:
[tx] updated from transifex

mtillberg committed:
fix/optimize preview cleanup query

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

DeepDiver1975 committed:
fix: Do not allow to set higher permissions on a federated share for a resource which was shared with limited permissions

dependabot[bot] committed:
Bump guzzlehttp/guzzle from 7.6.1 to 7.7.0

Bumps [guzzlehttp/guzzle](https://github.com/guzzle/guzzle) from 7.6.1 to 7.7.0.
- [Release notes](https://github.com/guzzle/guzzle/releases)
- [Changelog](https://github.com/guzzle/guzzle/blob/7.7/CHANGELOG.md)
- [Commits](https://github.com/guzzle/guzzle/compare/7.6.1...7.7.0)

---
updated-dependencies:
- dependency-name: guzzlehttp/guzzle
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump nikic/php-parser from 4.15.4 to 4.15.5

Bumps [nikic/php-parser](https://github.com/nikic/PHP-Parser) from 4.15.4 to 4.15.5.
- [Release notes](https://github.com/nikic/PHP-Parser/releases)
- [Changelog](https://github.com/nikic/PHP-Parser/blob/4.x/CHANGELOG.md)
- [Commits](https://github.com/nikic/PHP-Parser/compare/v4.15.4...v4.15.5)

---
updated-dependencies:
- dependency-name: nikic/php-parser
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

phil-davis committed:
changelog for guzzle 7.7.0

phil-davis committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

DeepDiver1975 committed:
fix: verify tag name length upon creation (#40804)

DeepDiver1975 committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Use symfony/translation 5.4 in acceptance tests

mrow4a committed:
make sure to use json for federation post and get (#40815)

* make sure to use json for federation post and get

* add changelog

* minor fixes

mrow4a committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Add acceptance tests for occ previews:cleanup

dependabot[bot] committed:
Bump symfony/process from 5.4.23 to 5.4.24

Bumps [symfony/process](https://github.com/symfony/process) from 5.4.23 to 5.4.24.
- [Release notes](https://github.com/symfony/process/releases)
- [Changelog](https://github.com/symfony/process/blob/6.3/CHANGELOG.md)
- [Commits](https://github.com/symfony/process/compare/v5.4.23...v5.4.24)

---
updated-dependencies:
- dependency-name: symfony/process
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

phil-davis committed:
PHP dependencies and changelog

phil-davis committed:
Automated changelog update [skip ci]

mmattel committed:
Update Google.php (cover variable initialisation for the else statement)

Fixes: https://github.com/owncloud/core/issues/40802 (Deleting a file in a Google Drive Mount throws an error)

There can be the case that the varible is not defined when passing the else statement throwing an `undefined variable` error. This is now fixed as both variables are now initialized.

phil-davis committed:
Automated changelog update [skip ci]

mmattel committed:
add changelog

mmattel committed:
fix changelog title length

phil-davis committed:
Automated changelog update [skip ci]

jvillafanez committed:
Add commands to handle the trusted servers from command line (#40796)

* Add commands to handle the trusted servers from command line

* Add changelog entry

* Add unit tests

* Improve messages

* Add additional message

jvillafanez committed:
Automated changelog update [skip ci]

phil-davis committed:
Handle possible null original-location in tryToDeleteFileFromTrashbin

phil-davis committed:
Bump PHP dependencies

phil-davis committed:
Automated changelog update [skip ci]

IljaN committed:
Update ca-bundle.crt

jnweiger committed:
bump version in sonar...

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
bump version in sonar-project to 10.13.0

phil-davis committed:
move changelog items for 10.12.2

phil-davis committed:
Automated changelog update [skip ci]

jvillafanez committed:
Enforce 2fa with exclusion groups

jvillafanez committed:
Ensure the requested open mode is set in the checksum wrapper

jvillafanez committed:
Adjust tests

phil-davis committed:
Automated changelog update [skip ci]

jvillafanez committed:
Add changelog entry

jvillafanez committed:
Adjust and add tests

jvillafanez committed:
Add changelog entry

phil-davis committed:
Bump sabre/uri (2.3.2 => 2.3.3)

phil-davis committed:
skip previews:cleanup acceptance test on older oC10

phil-davis committed:
Automated changelog update [skip ci]

jvillafanez committed:
Include comment and adjust panel priority

jvillafanez committed:
Typos in the template

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Bump phpseclib/phpseclib (3.0.19 => 3.0.20)

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Update php-jwt and some other files_external dependencies

jvillafanez committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Automated changelog update [skip ci]

amrita-shrestha committed:
Add scroll into view action (#40842)

phil-davis committed:
refactor isNewPermissionHigher (#40829)

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

phil-davis committed:
Fix handling of logger calls in lib/private/Log.php

phil-davis committed:
Add acceptance test for background:queue:execute

phil-davis committed:
Throw an exception if no logger method is available

phil-davis committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

navid-shokri committed:
Update ProviderFactory.php

IljaN committed:
Fix error when version_metadata is enabled and there is no quota (#40847)

* Fix error when version_metadata is enabled and there is no quota

* changelog

IljaN committed:
Automated changelog update [skip ci]

phil-davis committed:
Bump PHP dependencies

phil-davis committed:
Automated changelog update [skip ci]

navid-shokri committed:
Merge pull request #16 from pondersource/Fix#19-ocm

return defauldt Provider for Share_Type_Remote_Group

phil-davis committed:
Apply code-style changes from php-cs-fixer 3.20

phil-davis committed:
Bump sabre/xml (2.2.5 to 2.2.6)

phil-davis committed:
Automated changelog update [skip ci]

phil-davis committed:
Bump owncloud/tarstreamer (2.0.0 to 2.1.0)

phil-davis committed:
Automated changelog update [skip ci]

ownclouders committed:
[tx] updated from transifex

michielbdejong commented:
@phil-davis all checks have passed! :)

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
[tx] updated from transifex

DeepDiver1975 committed:
fix: check for empty string if header exists

phil-davis commented:
I made a separate PR with this code rebased to current master. That demonstrates that the CI passes right today.

@DeepDiver1975 @jvillafanez please review. I  can't really review this effectively because I am not familiar enough with exactly what core "hook points" are needed.

jvillafanez commented on apps/files_sharing/lib/Controllers/ExternalSharesController.php:
```diff
@@ -77,21 +91,33 @@ public function __construct(
-	public function create($id) {
-		$shareInfo = $this->externalManager->getShare($id);
+	public function create($id, $share_type) {
+		if ($share_type === "group" && $this->groupExternalManager !== null) {
```
You've already have a `getManagerForShareType` method.

jvillafanez commented on apps/files_sharing/lib/Controllers/ExternalSharesController.php:
```diff
@@ -131,11 +158,20 @@ public function destroy($id) {
 	}
 
+	private function getManagerForShareType($share_type) {
+		if ($share_type === "group" && $this->groupExternalManager !== null) {
```
It seems better to compare the share type against a constant.

jvillafanez commented on lib/private/Share20/Manager.php:
```diff
@@ -1529,8 +1530,25 @@ public function getShareByToken($token) {
+			}
+		}
+
+		if ($share === null) {
```
What about moving this block inside the `catch (ShareNotFound $ex)`?

jvillafanez commented on lib/private/Share20/Manager.php:
```diff
@@ -1529,8 +1530,25 @@ public function getShareByToken($token) {
+				if ($provider !== null) {
+					$share = $provider->getShareByToken($token);
+				}
+			} catch (ProviderException $ex) {
```
This requires either a comment or a log.

jvillafanez reviewed (commented):
In general, for the js code all the sentences should end with `;` for consistency

jvillafanez commented:
Not sure who can test this, but we need to ensure there are no problems with this new code with and without the app.

phil-davis commented:
"without the app" should be OK - the full CI test suite passes, including various federated API acceptance tests etc. So there is not any regression found when running a system without the app and without `sharing.ocmController` set.

If someone sets `sharing.ocmController` but does not install the/an app that has code that provides things like a `createShare` method, then there will be trouble. But I suppose this is a "plugin" that would only be "supported" for installations that are managing this stuff.

phil-davis committed:
Remove cancelPreviousBuilds pipeline from drone

phil-davis committed:
Remove early-fail stop-build step from acceptance test pipelines

ownclouders committed:
[tx] updated from transifex

jvillafanez committed:
Improve query performance

jvillafanez committed:
Add changelog entry

phil-davis committed:
Adjust drone env variables for changes to drone CI

phil-davis committed:
Fix references to CACHE_BUCKET

ownclouders committed:
[tx] updated from transifex

DeepDiver1975 committed:
fix: no CSRF required on GET requests

Commit:
Skip share_folder for guest users (#40864)

* skip share_folder for guest users

* add changelog

* appy suggestions

* update config.sample.php

* fix return

* just return root

mmattel committed:
Update config.apps.sample.php

DeepDiver1975 committed:
fix: no CRSF on /privatedata/getattribute

Commit:
Disable online updater for enterprise (#40841)

* disable online updater for enterprise

* add changelog entry

* fix changelog entry

* add changelog entry

dependabot[bot] committed:
Bump semver from 6.3.0 to 6.3.1 in /build

Bumps [semver](https://github.com/npm/node-semver) from 6.3.0 to 6.3.1.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v6.3.1/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v6.3.0...v6.3.1)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

phil-davis committed:
Fix indent that php-style does not like now

dependabot[bot] committed:
Bump phpunit/phpunit from 9.6.9 to 9.6.10

Bumps [phpunit/phpunit](https://github.com/sebastianbergmann/phpunit) from 9.6.9 to 9.6.10.
- [Changelog](https://github.com/sebastianbergmann/phpunit/blob/9.6.10/ChangeLog-9.6.md)
- [Commits](https://github.com/sebastianbergmann/phpunit/compare/9.6.9...9.6.10)

---
updated-dependencies:
- dependency-name: phpunit/phpunit
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump phpseclib/phpseclib from 3.0.20 to 3.0.21

Bumps [phpseclib/phpseclib](https://github.com/phpseclib/phpseclib) from 3.0.20 to 3.0.21.
- [Release notes](https://github.com/phpseclib/phpseclib/releases)
- [Changelog](https://github.com/phpseclib/phpseclib/blob/master/CHANGELOG.md)
- [Commits](https://github.com/phpseclib/phpseclib/compare/3.0.20...3.0.21)

---
updated-dependencies:
- dependency-name: phpseclib/phpseclib
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

phil-davis committed:
changelog for phpseclib 3.0.21

phil-davis committed:
Remove github comment from drone CI

phil-davis committed:
Remove unused drone-github-comment image

ownclouders committed:
[tx] updated from transifex

ownclouders committed:
Automated changelog update [skip ci]

phil-davis committed:
Automated changelog update [skip ci]

soltanireza65 commented on apps/files_sharing/js/app.js:
```diff
@@ -221,6 +221,9 @@ OCA.Sharing.App = {
 			dataType: 'json',
 			type: method,
+			data: JSON.stringify({
+				...(!!shareType && { shareType }),
```
this is an es6+ syntax
the requests body needs to be in a stringified js object
what this line is trying to do is :
check if `shareType` exists make an object as : 
```json
{"shareType" : shareType} 
```
and pass it to JSON.stringify and then pass the stringified object as a body
so if shareType does not exist body would be an empty object

soltanireza65 replied:
it's the same as:
```javascript
JSON.stringify((Boolean(shareType) ? { shareType: shareType } : {}))
```

shokri-navid committed:
improve logics

shokri-navid committed:
improve logics

shokri-navid committed:
Resolve PR review conversations

shokri-navid committed:
adjust code format

shokri-navid committed:
remove extra space

shokri-navid committed:
remove extra space

shokri-navid committed:
remove extra space

navid-shokri committed:
Merge pull request #17 from pondersource/fix/review-comments

Fix/review comments

shokri-navid committed:
add logger to RemoteOcsController

shokri-navid committed:
resolve RemotOcsControllerTest error

Commit:
improve log messages for encryption rename (#40875)

jvillafanez commented on apps/files_sharing/js/app.js:
```diff
@@ -209,18 +209,18 @@ OCA.Sharing.App = {
 			contentType: 'application/json',
 			dataType: 'json',
 			type: method,
+			data: JSON.stringify((Boolean(shareType) ? { shareType: shareType } : {})),
```
Isn't it better `(shareType === undefined ? ......)`?
I'd expect the shareType to be an integer (1,2,3....) or a string (user, group, remote...), so converting them to a boolean seems weird, specially when it isn't the main case. If the only additional case we need to handle is the "undefined", then checking for `shareType === undefined` makes it clearer.
If there are additional cases that I'm not aware of, we might need a code comment so we know what can be set in the shareType parameter.

soltanireza65 replied:
if the shareType happens to be an empty string (for whatever reason) it will not result in undefined, so the condition may not work as expected
so the base case is to cast conditions to bool unless we are 100% sure

jvillafanez replied:
I'm not fully convinced...

If we consider the empty string as invalid in order to not send it to the server, we should also include strings such as "aaa" or "bbb" which don't make sense for a share type.
It seems something that should be handled in the server, so if the client sends an empty share type (or any other invalid type), the server can reject the request (or assume a default share type, whatever is more convenient).

I think the best compromise is to include a code comment explaining the expectations, so we know that we don't want to send an empty string as shareType.

shokri-navid committed:
Merge remote-tracking branch 'refs/remotes/origin/accept-ocm-to-groups' into accept-ocm-to-groups

Commit:
check if shareType is undefined

Commit:
add comment to apps/files_sharing/api/v1 body `shareType`

jnweiger committed:
Bump minimum.supported.desktop.version from 2.3.3 to 3.2.1 (#40876)

as suggested in https://github.com/owncloud/enterprise/issues/5859#issuecomment-1634463398

shokri-navid committed:
convert string share type to int

phil-davis commented:
https://drone.owncloud.com/owncloud/core/38743/5/1 cancel-previous-builds is no longer part of drone CI, due to changes to drone secrets in the last week or two.

Please rebase this PR to latest master, then the CI should work again.

shokri-navid committed:
rebase to upstream master

shokri-navid committed:
code style fix

phil-davis commented:
Now there are 220 commits in this PR - what happened???

shokri-navid committed:
add missing logger parameter in RemoteOcsController

T0mWz commented:
When appling this patch, I got;
```
Exception: No share provider for share type 7
```

Maybe good to verify behavior without Sciencemesh app, or it should be default shipped.

phil-davis commented:
> Now there are 220 commits in this PR - what happened???

I cherry-picked what I think are the real commits for this PR and created PR #40877 - that should be easier to review, rather than whatever happened here to have 221 commits listed.

jvillafanez commented:
Left a comment in https://github.com/owncloud/core/pull/40877/files#r1265137966 , but for the rest the other PR looks good.

> When appling this patch, I got;
Exception: No share provider for share type 7
Maybe good to verify behavior without Sciencemesh app, or it should be default shipped.

Maybe a problem in https://github.com/owncloud/core/pull/40877/files#diff-6c825986578290fe355bffe05c5232629a36e554d48f5d889bb2c1ec73645265R1542
I think we'll have to check what providers are available before trying to get an optional one.

> Now there are 220 commits in this PR - what happened???

We need to decide how to move forward. Options are:
* They fix the branch on their side
* They open a new PR with the relevant changes. Basically https://github.com/owncloud/core/pull/40877 but on their side so they can keep applying the changes needed.
* We take over the PR from https://github.com/owncloud/core/pull/40877 . This doesn't seem a good solution because I don't think we can test the whole solution, just that it doesn't break if the app isn't installed.

phil-davis commented:
@michielbdejong @shokri-navid 
was this done in #40886 ?

Can we close this PR?

jvillafanez commented:
I think this is the old PR that got broken and was superseded by the one that was already merged, but we can wait for confirmation.

michielbdejong commented:
Superseded by #40886
